sdk/cosmosdb/cosmos/src/request/RestError.ts

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+/**
+ * The options supported by RestError.
+ */
+export interface RestErrorOptions {
+  /**
+   * The error message.
+   */
+  message?: string;
+  /**
+   * A string error code from an underlying system call.
+   */
+  code?: string;
+  /**
+   * The HTTP status code of the request (if applicable.)
+   */
+  statusCode?: number;
+  /**
+   * The request that was made.
+   */
+  request?: unknown;
+  /**
+   * The response received (if any.)
+   */
+  response?: unknown;
+  /**
+   * Bonus property set by the throw site.
+   */
+  details?: unknown;
+}
+
+/**
+ * A custom error type for failed pipeline requests.
+ */
+export class RestError extends Error {
```

Comment:
May I ask why you are wrapping this error with your own version? Most SDKs simply re-export `RestError` from core-rest-pipeline directly. 

One small advantage (besides saving the original callstack) of using our Error type is we do try to make it a little harder to leak sensitive data: https://github.com/Azure/azure-sdk-for-js/blob/d83b1ddf0c363df02768bbb9922377d39ab3c95d/sdk/core/core-rest-pipeline/src/restError.ts#L85

Reply:
@xirzec Could you please provide any example where `RestError` is exported from rest-pipeline directly?

Reply:
Sorry for the delay here over the holidays. I think you are handling it well in #24373, but did you want additional examples?

Reply:
@xirzec Thanks for suggesting the re-exporting errors approach.